src/integer/mat_z/set.rs

```diff
@@ -62,6 +78,164 @@ implement_for_owned!(Z, MatZ, SetEntry);
 
 implement_for_others!(Z, MatZ, SetEntry for i8 i16 i32 i64 u8 u16 u32 u64);
 
+impl MatZ {
+    /// Swaps two entries of the specified matrix.
+    ///
+    /// Parameters:
+    /// - `row0`: specifies the row, in which the first entry is located
+    /// - `col0`: specifies the column, in which the first entry is located
+    /// - `row1`: specifies the row, in which the second entry is located
+    /// - `col1`: specifies the column, in which the second entry is located
+    ///
+    /// Returns an empty `Ok` if the action could be performed successfully.
+    /// Otherwise, a [`MathError`] is returned if one of the specified entries is not part of the matrix.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer::MatZ;
+    ///
+    /// let mut matrix = MatZ::new(4, 3).unwrap();
+    /// matrix.swap_entries(0, 0, 2, 1);
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`MathError::OutOfBounds`]
+    /// if the number of rows or columns is greater than the matrix or negative.
+    pub fn swap_entries(
+        &mut self,
+        row0: impl TryInto<i64> + Display,
+        col0: impl TryInto<i64> + Display,
+        row1: impl TryInto<i64> + Display,
+        col1: impl TryInto<i64> + Display,
+    ) -> Result<(), MathError> {
+        let (row0, col0) = evaluate_indices(self, row0, col0)?;
+        let (row1, col1) = evaluate_indices(self, row1, col1)?;
+
+        let entry0 = self.get_entry(row0, col0)?;
+        let entry1 = self.get_entry(row1, col1)?;
+
+        self.set_entry(row0, col0, entry1)?;
+        self.set_entry(row1, col1, entry0)?;
```

Comment:
should there be a more efficient possible implementation, where we can use the FLINT methods for this, using fmpz_mat_entry and then just swapping the values (i.e. the pointer or respectively the value). It is not as important for Z, but looking forward to other structs like MatPolyZ or similiar this could be a great imporvement.

If you do not want to improve it now, just add a todo or similar.